applications.js

```diff
@@ -0,0 +1,61 @@
+const DatArchive = require('node-dat-archive')
+const { lstatSync, readdirSync } = require('fs')
+const { join, basename } = require('path')
+const mkdirp = require('mkdirp')
+const sleep = require('sleep-promise')
+
+const APPLICATIONS_DIR = join(__dirname, 'applications')
+
+const isDirectory = source => lstatSync(source).isDirectory()
+
+const getDirectories = source =>
+  readdirSync(source)
+    .map(name => join(source, name))
+    .filter(isDirectory)
+    .map(name => basename(name))
+    .filter(name => name.length === 64)
+
+mkdirp.sync(APPLICATIONS_DIR)
+
+const archives = new Map()
+
+exports.getApplicationPubs = () => archives
+
+exports.createApplicationPub = async applicationKey => {
+  mkdirp.sync(join(APPLICATIONS_DIR, applicationKey, 'pub'))
+
+  const archive = await DatArchive.create({
+    localPath: join(APPLICATIONS_DIR, applicationKey, 'pub'),
+    title: `Users of dat://${applicationKey}`,
+    description: 'User pub for application',
+    datOptions: { latest: true }
+  })
+
+  // if I load prematurely, it doesn't work
+  sleep(3000).then(loadAllArchives)
```

Comment:
Should there be an `await` here?

Reply:
No, dat location can be immediately returned, even before the hub begins to successfully host it.